Added negative scan increment compatibility

felix-kolbe opened the pull request:
And moved duplicated loopwise incrementing line

fuerte backport (cherry picked from commit 1bb7353deff061dc0e5b0b0ea477b1e5fa370668)

ps: If this isn't the way to backport I'd appreciate any hints.

Commit:
Added negative scan increment compatibility

And moved duplicated loopwise incrementing line

fuerte backport (cherry picked from commit 1bb7353deff061dc0e5b0b0ea477b1e5fa370668)

vrabaud commented on include/laser_filters/angular_bounds_filter.h:
```diff
@@ -74,9 +74,9 @@
           //wait until we get to our desired starting angle
-          if(start_angle < lower_angle_){
+          if((input_scan.angle_increment > 0 && start_angle < lower_angle_) || 
+              (input_scan.angle_increment < 0 && start_angle > lower_angle_)){
```
I disagree: the goal of that condition is to get inside the  [ lower_angle_ , upper_angle_ ] interval.If input_scan.angle_increment < 0 , then this just sets us right outside and below the interval.
I believe the right condition should be: if((start_angle < lower_angle_) || (start_angle > upper_angle_))

vrabaud commented on include/laser_filters/angular_bounds_filter.h:
```diff
@@ -89,13 +89,13 @@
             //check if we need to break out of the loop, basically if the next increment will put us over the threshold
-            if(current_angle + input_scan.angle_increment > upper_angle_){
+            if((input_scan.angle_increment > 0 && current_angle + input_scan.angle_increment > upper_angle_) || 
+                (input_scan.angle_increment < 0 && current_angle + input_scan.angle_increment < upper_angle_)){
```
Similarly here :  if((current_angle + input_scan.angle_increment < lower_angle_) || (current_angle + input_scan.angle_increment > upper_angle_))

Commit:
Better solution of negative increment compatibility.

The previous commit assumes a laser_filter configuration with lower_angle > upper_angle, which wouldn't be inter robot compatible. Now lower_angle is expected to be smaller than upper_angle even if the increment is negative.

Commit:
Word fix in comments: rays, not scans
